lib/web_ui/lib/src/engine/canvaskit/color_filter.dart

```diff
@@ -5,62 +5,147 @@
+
+  @override
+  SkColorFilter resurrect() => ckColorFilter._initRawColorFilter();
+
+  @override
+  void delete() {
+    rawSkiaObject?.delete();
   }
 
   @override
-  SkColorFilter createDefault() {
-    return _createSkiaObjectFromFilter();
+  int get hashCode => ckColorFilter.hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    if (runtimeType != other.runtimeType)
+      return false;
+    return other is _ManagedSkColorFilter
+        && other.ckColorFilter == ckColorFilter;
   }
 
   @override
-  SkColorFilter resurrect() {
-    return _createSkiaObjectFromFilter();
+  String toString() => ckColorFilter.toString();
+}
+
+/// A [ui.ColorFilter] backed by Skia's [SkColorFilter].
+///
+/// Additionally, this class provides the interface for converting itself to a
+/// [ManagedSkiaObject] that manages a skia image filter.
+abstract class CkColorFilter implements _CkManagedSkImageFilterConvertible<SkImageFilter>, EngineColorFilter {
+  const CkColorFilter();
+
+  /// Called by [ManagedSkiaObject.createDefault] and
+  /// [ManagedSkiaObject.resurrect] to create a new [SKImageFilter], when this
+  /// filter is used as an [ImageFilter].
+  SkImageFilter _initRawImageFilter() => canvasKit.SkImageFilter.MakeColorFilter(_initRawColorFilter(), null);
+
+  /// Called by [ManagedSkiaObject.createDefault] and
+  /// [ManagedSkiaObject.resurrect] to create a new [SKColorFilter], when this
+  /// filter is used as a [ColorFilter].
+  SkColorFilter _initRawColorFilter();
+
+  ManagedSkiaObject<SkImageFilter> get _imageFilter => _CkColorFilterImageFilter(colorFilter: this);
+}
+
+class _CkBlendModeColorFilter extends CkColorFilter {
+  const _CkBlendModeColorFilter(this.color, this.blendMode);
+
+  final ui.Color color;
+  final ui.BlendMode blendMode;
+
+  @override
+  SkColorFilter _initRawColorFilter() {
+    return canvasKit.SkColorFilter.MakeBlend(
+      toSharedSkColor1(color),
+      toSkBlendMode(blendMode),
+    );
   }
 
   @override
-  void delete() {
-    rawSkiaObject?.delete();
+  int get hashCode => ui.hashValues(color, blendMode);
+
+  @override
+  bool operator ==(Object other) {
+    if (runtimeType != other.runtimeType)
+      return false;
+    return other is _CkBlendModeColorFilter
+        && other.color == color
+        && other.blendMode == blendMode;
+  }
+
+  @override
+  String toString() => 'ColorFilter.mode($color, $blendMode)';
+}
+
+class _CkMatrixColorFilter extends CkColorFilter {
+  const _CkMatrixColorFilter(this.matrix);
+
+  final List<double> matrix;
+
+  @override
+  SkColorFilter _initRawColorFilter() {
+    assert(this.matrix.length == 20,  'Color Matrix must have 20 entries.');
+    final List<double> matrix = this.matrix;
+    if (matrix is Float32List)
+      return canvasKit.SkColorFilter.MakeMatrix(matrix);
+    final Float32List float32Matrix = Float32List(20);
+    for (int i = 0; i < 20; i++) {
+      float32Matrix[i] = matrix[i]
```

Comment:
Missing semicolon is causing the builds to fail.

Reply:
whoops. thanks for catching this!